src/uucore/src/lib/features/selinux.rs

```diff
@@ -328,6 +371,187 @@ pub fn preserve_security_context(from_path: &Path, to_path: &Path) -> Result<(),
     set_selinux_security_context(to_path, Some(&context))
 }
 
+/// Helper function to collect paths that need SELinux context setting.
+///
+/// Traverses from the given root path up to existing parent directories.
+/// Returns a vector of paths in reverse order (from parent to child).
+fn collect_paths_for_context_setting(root_path: &Path) -> Vec<&Path> {
+    let mut paths_to_set = Vec::new();
+    let mut current_path = root_path;
+
+    while current_path != Path::new("") && current_path != Path::new("/") && current_path.exists() {
+        paths_to_set.push(current_path);
+        if let Some(parent) = current_path.parent() {
+            if parent == current_path {
+                break;
+            }
+            current_path = parent;
+        } else {
+            break;
+        }
+    }
```

Comment:
I think you can simplify it with using the `ancestors` function, something like:
```rust 
    let mut paths: Vec<&Path> = root_path
        .ancestors()
        .take_while(|p| p.exists())
        .collect();
```